doc/developer/grpc.rst

```diff
@@ -17,14 +28,274 @@ next step is to generate the FRR northbound bindings. To generate the
+  				       grpc::InsecureChannelCredentials());
+      auto stub = frr::Northbound::NewStub(channel);
+
+      request.set_type(frr::GetRequest::ALL);
+      request.set_encoding(frr::JSON);
+      request.set_with_defaults(true);
+      request.add_path("/frr-interface:lib");
+      auto stream = stub->Get(&context, request);
+
+      std::ostringstream ss;
+      while (stream->Read(&reply))
+        ss << reply.data().data() << std::endl;
+
+      status = stream->Finish();
+      assert(status.ok());
+      std::cout << "Interface Info:\n" << ss.str() << std::endl;
+  }
+
+Here is how to compile and run the program, with the example output:
+
+::
+
+  $ g++ -o test test.cpp frr-northbound.grpc.pb.cc frr-northbound.pb.cc -lgrpc++ -lprotobuf
+  $ ./test
+  Interface Info:
+  {
+    "frr-interface:lib": {
+      "interface": [
+        {
+          "name": "lo",
+          "vrf": "default",
+          "state": {
+            "if-index": 1,
+            "mtu": 0,
+            "mtu6": 65536,
+            "speed": 0,
+            "metric": 0,
+            "phy-address": "00:00:00:00:00:00"
+          },
+          "frr-zebra:zebra": {
+            "state": {
+              "up-count": 0,
+              "down-count": 0,
+              "ptm-status": "disabled"
+            }
+          }
+        },
+        {
+          "name": "r1-eth0",
+          "vrf": "default",
+          "state": {
+            "if-index": 2,
+            "mtu": 1500,
+            "mtu6": 1500,
+            "speed": 10000,
+            "metric": 0,
+            "phy-address": "02:37:ac:63:59:b9"
+          },
+          "frr-zebra:zebra": {
+            "state": {
+              "up-count": 0,
+              "down-count": 0,
+              "ptm-status": "disabled"
+            }
+          }
+        }
+      ]
+    },
+    "frr-zebra:zebra": {
+      "mcast-rpf-lookup": "mrib-then-urib",
+      "workqueue-hold-timer": 10,
+      "zapi-packets": 1000,
+      "import-kernel-table": {
+        "distance": 15
+      },
+      "dplane-queue-limit": 200
+    }
+  }
+
+
+
+.. _grpc-python-example:
+
+Python Example
+--------------
+
+The next sections will use Python as an example for writing scripts to use
 the northbound.
 
+.. _grpc-python-generate:
+
+Generating Python FRR Bindings
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Generating FRR northbound bindings for Python example:
+
+::
+
+   # Install python3 virtual environment capability e.g.,
+   sudo apt-get install python3-venv
+
+   # Create a virtual environment for python grpc and activate
+   python3 -m venv venv-grpc
+   source venv-grpc/bin/activate
+
+   # Install grpc requirements
+   pip install grpcio grpcio-tools
+
+   mkdir /tmp/frr-python
+   cd grpc
+
+   python3 -m grpc_tools.protoc  \
+           --python_out=/tmp/frr-python \
+           --grpc_python_out=/tmp/frr-python \
+           -I $(pwd) \
+           frr-northbound.proto
+
+.. _grpc-ruby-if-sample:
```

Comment:
"ruby"?

Reply:
indeed, supposed to be python, pushed a fix along with more descriptions of these sections and what they show.